metricflow/model/semantics/semantic_containers.py

```diff
@@ -328,3 +328,8 @@ def get_aggregation_time_dimensions_with_measures(
             data_source_reference in self._data_source_to_aggregation_time_dimensions
         ), f"Data Source {data_source_reference} is not known"
         return self._data_source_to_aggregation_time_dimensions[data_source_reference]
+
+    def get_data_sources_for_identifier(self, identifier_reference: IdentifierReference) -> List[DataSource]:
+        """Return all data sources associated with and identifier reference"""
```

Comment:
We should either document that this might return multiple pointers to a data source, or else alter the signature in the Protocol to return a distinct set (either `Set[DataSource]` or else `Dict[str, DataSource]` where the str represents the data source name).